internal/installercache/installercache_test.go

```diff
@@ -52,16 +108,27 @@ var _ = Describe("installer cache", func() {
 				err := os.WriteFile(fname, []byte("abcde"), 0600)
 				return "", err
 			})
-		l, err := manager.Get(releaseID, "mirror", "pull-secret", mockRelease, version)
+		l, err := manager.Get(ctx, releaseID, "mirror", "pull-secret", mockRelease, version, clusterID)
 		Expect(err).ShouldNot(HaveOccurred())
+		Expect(l.releaseID).To(Equal(releaseID))
+		Expect(l.clusterID).To(Equal(clusterID))
+		Expect(l.startTime).ShouldNot(BeZero())
+		Expect(l.cached).To(Equal(expectCached))
+		Expect(l.eventsHandler).To(Equal(eventsHandler))
+		if !expectCached {
+			Expect(l.extractDuration).ShouldNot(BeZero())
+		}
+		Expect(l.Path).ShouldNot(BeEmpty())
 
 		time.Sleep(1 * time.Second)
```

Comment:
I know this is already here, but why is this needed?

Reply:
Actually, that's a very interesting point. 

I think what they were trying to do was simulate the passage of time. The way we structure calls to `time.Now()` and such (throughout the application and not just here) does not make them very testable. The same applies to filesystem calls such as `os.Stat(..)`

Ideally, we should wrap calls to "non mockabe" calls in an interface that can be mocked. 

This would allow us to avoid all of this awkwardness and test things like filesystem errors or the advancement of time without hacks like this.

I personally think that we need to address this as part of the installer cache fixes and I was planning on this.
But that it does not need to be done in this PR.